src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -86,4 +101,181 @@ object Playlist {
 			else add(track)
 		}
 	}
+}
+
+object PlaylistManager {
+	suspend fun loadPlaylist(apiConnection: APIConnection){
+		val tracks = apiConnection.getTracks()
+		tracks?.forEachIndexed { index, track ->
+			val found = Cache.getTracks().find {
+				it.id == track.id
+			}
+			if (found != null)
+				tracks[index] = found
+			else
+				tracks.removeAt(index)
```

Comment:
Shouldn't you log this? Seems like an unexpected case for me.

Reply:
335b23e338886d68ad78d2ec64d6b79b7e999a88